avldb/src/main/scala/scorex/db/LDBFactory.scala

```diff
@@ -138,15 +137,25 @@ object LDBFactory extends ScorexLogging {
 
   lazy val factory: DBFactory = {
     val loaders = List(ClassLoader.getSystemClassLoader, this.getClass.getClassLoader)
-    val factories = List(nativeFactory, javaFactory)
+
+    // As LevelDB-JNI has problems on Mac (see https://github.com/ergoplatform/ergo/issues/1067),
+    // we are using only pure-Java LevelDB on Mac
+    val isMac = System.getProperty("os.name").toLowerCase().indexOf("mac") >= 0
```

Comment:
I suggest to google for a more principled way of detecting macOS and leave the reference in the code.

Reply:
@aslesarenko there's ready check in Apache common-utils, but this library is not available in the avldb project, and no need to get a dependency for 1 line I guess. I checked an implementation there also, it works like the line of code of mine basically.